src/amp/lib/ad-json.test.ts

```diff
@@ -86,7 +89,9 @@ describe('ampadslots', () => {
 		if (p === undefined) {
 			return fail();
 		}
-		expect(p.value).toBe('4,5,1,2,3');
+		// These expect statements are returned to keep typescript happy
+		// Otherwise it complains that not all code paths return a value
+		return expect(p.value).toBe('4,5,1,2,3');
```

Comment:
Thinking about these cases more, what about just removing the return from the `fail()` calls and using if/else. E.g

```
		if (p === undefined) {
			fail();
		} else {
			expect(p.value).toBe('amp');
		}
```

It would avoid the need for returns (which are not doing anything here as no consumer/jest works by side-effects). And also we could then remove the comments.

```
		if (p === undefined) {
			fail();
		}
		
		// imagine some code here
		expect(p.value).toBe('amp');
```

also works but in the `// imagine...` bit stuff would actually get executed even on a fail, which could be confusing/slow things down. So the if/else feels better here.

Reply:
Sounds good to me 👍🏻